src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -39,29 +40,29 @@ import           ZkFold.Symbolic.MonadCircuit               (Arithmetic)
     Additionally, we don't want this library to depend on Cardano libraries.
 -}
 
-data Plonk (n :: Natural) (l :: Natural) curve1 curve2 transcript = Plonk {
+data Plonk (n :: Natural) (l :: Natural) curve1 curve2 transcript core =
+    CoreFunction curve1 core =>
```

Comment:
Are you sure we don't want to add core only to `NonInteractiveProof`? 🤔

Reply:
Yes, you are right. Here I can remove the core from the definition and leave it only in the constraint for the NonInteractiveProof instance